lib/src/Uncertainty/Distribution/TruncatedDistribution.cxx

```diff
@@ -227,74 +204,82 @@ Point TruncatedDistribution::getRealization() const
   return normalizationFactor_ * distribution_.computePDF(point);
 }
 
 
 /* Get the CDF of the distribution: CDF_trunc = 1[a, b] * (CDF - CDF(a)) / (CDF(b) - CDF(a)) + 1]b, inf] */
 Scalar TruncatedDistribution::computeCDF(const Point & point) const
 {
-  if (point.getDimension() != 1) throw InvalidArgumentException(HERE) << "Error: the given point must have dimension=1, here dimension=" << point.getDimension();
+  if (point.getDimension() != getDimension()) throw InvalidArgumentException(HERE) << "Error: the given point must have dimension=" << getDimension() << ", here dimension=" << point.getDimension();
+
+  for (UnsignedInteger k = 0; k < getDimension(); ++ k)
+  {
+    const Scalar xK = point[k];
+    if (xK <= bounds_.getLowerBound()[k]) return 0.0;
+    if (xK >= bounds_.getUpperBound()[k]) return 1.0;
+  }
 
-  const Scalar x = point[0];
-  if (x <= lowerBound_) return 0.0;
-  if (x >= upperBound_) return 1.0;
   // If tail=true, don't call distribution_.computeCDF with tail=true in the next line!
   return normalizationFactor_ * (distribution_.computeCDF(point) - cdfLowerBound_);
 }
 
 Scalar TruncatedDistribution::computeComplementaryCDF(const Point & point) const
 {
-  if (point.getDimension() != 1) throw InvalidArgumentException(HERE) << "Error: the given point must have dimension=1, here dimension=" << point.getDimension();
+  if (point.getDimension() != getDimension()) throw InvalidArgumentException(HERE) << "Error: the given point must have dimension=" << getDimension() << ", here dimension=" << point.getDimension();
+
+  for (UnsignedInteger k = 0; k < getDimension(); ++ k)
+  {
+    const Scalar xK = point[k];
+    if (xK <= bounds_.getLowerBound()[k]) return 1.0;
+    if (xK >= bounds_.getUpperBound()[k]) return 0.0;
+  }
 
-  const Scalar x = point[0];
-  if (x <= lowerBound_) return 1.0;
-  if (x > upperBound_) return 0.0;
   // If tail=true, don't call distribution_.computeCDF with tail=true in the next line!
   return normalizationFactor_ * (cdfUpperBound_ - distribution_.computeCDF(point));
 }
 
 /* Get the PDFGradient of the distribution */
 Point TruncatedDistribution::computePDFGradient(const Point & point) const
 {
-  if (point.getDimension() != 1) throw InvalidArgumentException(HERE) << "Error: the given point must have dimension=1, here dimension=" << point.getDimension();
+  if (point.getDimension() != getDimension()) throw InvalidArgumentException(HERE) << "Error: the given point must have dimension=1, here dimension=" << point.getDimension();
 
   const Scalar x = point[0];
-  if ((x <= lowerBound_) || (x > upperBound_)) return Point(distribution_.getParametersCollection()[0].getDimension() + finiteLowerBound_ + finiteUpperBound_);
+  if ((x <= bounds_.getLowerBound()[0]) || (x > bounds_.getUpperBound()[0])) return Point(distribution_.getParametersCollection()[0].getDimension() + bounds_.getFiniteLowerBound()[0] + bounds_.getFiniteUpperBound()[0]);
```

Comment:
Should check ALL the components of point, not only x

Reply:
I throw here if dimension is >1